app/views/browse/imagestream.html

```diff
@@ -72,20 +72,35 @@
                     <td data-title="From">
                       <!-- this value can get long when its an ImageStreamImage. Use max width and class truncate to elide some
                            of the SHA1 for long values. They can still be copied even when elided. -->
-                      <div style="max-width: 400px;" class="truncate">
+
                         <span ng-if="!tag.spec.from"><em>pushed</em></span>
-                        <span ng-if="tag.spec.from">
-                          <span ng-if="!tag.spec.from._imageStreamName">
-                            {{tag.spec.from.name}}
-                          </span>
-                          <span ng-if="tag.spec.from._imageStreamName">
-                            <span ng-if="tag.spec.from._imageStreamName === imageStream.metadata.name">{{tag.spec.from._completeName}}</span>
-                            <span ng-if="tag.spec.from._imageStreamName !== imageStream.metadata.name" >
-                              <a ng-href="{{tag.spec.from._imageStreamName | navigateResourceURL : 'ImageStream' : (tag.spec.from.namespace || imageStream.metadata.namespace)}}"><span ng-if="tag.spec.from.namespace && tag.spec.from.namespace !== imageStream.metadata.namespace" >{{tag.spec.from.namespace}}/</span>{{tag.spec.from._imageStreamName}}</a>{{tag.spec.from._nameConnector}}{{tag.spec.from._idOrTag}}
+                        <div ng-if="tag.spec.from">
+                          <!-- Wrap when < 768 -->
+                          <div class="word-break visible-xs-block">
+                            <span ng-if="!tag.spec.from._imageStreamName">
+                              {{tag.spec.from.name}}
                             </span>
-                          </span>
-                        </span>
-                      </div>
+                            <span ng-if="tag.spec.from._imageStreamName">
+                              <span ng-if="tag.spec.from._imageStreamName === imageStream.metadata.name">{{tag.spec.from._completeName}}</span>
+                              <span ng-if="tag.spec.from._imageStreamName !== imageStream.metadata.name" >
+                                <a ng-href="{{tag.spec.from._imageStreamName | navigateResourceURL : 'ImageStream' : (tag.spec.from.namespace || imageStream.metadata.namespace)}}"><span ng-if="tag.spec.from.namespace && tag.spec.from.namespace !== imageStream.metadata.namespace" >{{tag.spec.from.namespace}}/</span>{{tag.spec.from._imageStreamName}}</a>{{tag.spec.from._nameConnector}}{{tag.spec.from._idOrTag}}
+                              </span>
+                            </span>
+                          </div>
+                          <!-- Truncate when > 768 -->
+                          <div class="pull-spec truncate hidden-xs">
```

Comment:
If the markup is identical and only the classes change, can we use mixins and media queries to assign the styles for different screen widths? It's a little more logic than I'd like to repeat. I think this can be done entirely with less.